test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatConfigTests.cs

```diff
@@ -218,6 +219,117 @@ public void ConfigGetConfigKeyCommand_HelpMessage_Success()
+
+            var result = CommandRunner.Run(
+                DotnetCli,
+                Directory.GetCurrentDirectory(),
+                $"{XplatDll} config set {key} {value} --configfile {filePath}",
+                waitForExit: true);
+
+            ISettings settings = Configuration.Settings.LoadDefaultSettings(
+                testInfo.WorkingPath,
+                configFileName: null,
+                machineWideSettings: new XPlatMachineWideSetting());
+
+            var configSection = settings.GetSection("config");
+            var values = configSection.Items.Select(c => c as AddItem).Where(c => c != null).ToList();
+            var configItems = values.Where(i => i.Key == key);
+            var configFilePath = configItems.Single().ConfigPath;
+
+            // Assert
+            Assert.Equal(0, result.ExitCode);
+            Assert.Equal(value, configItems.Single().Value);
+            Assert.Equal(filePath, configFilePath);
+        }
+
+        [Fact]
+        public void ConfigSetCommand_WithConfigFileArg_UpdatesSetting()
+        {
+            // Arrange & Act
+            using var testInfo = new TestInfo("NuGet.Config");
+            var key = "http_proxy";
+            var value = "http://company-octopus:8765@contoso.test";
+            var filePath = Path.Combine(testInfo.WorkingPath, "NuGet.Config");
+
+            var result = CommandRunner.Run(
+                DotnetCli,
+                Directory.GetCurrentDirectory(),
+                $"{XplatDll} config set {key} {value} --configfile {filePath}",
+                waitForExit: true);
+
+            var settings = Configuration.Settings.LoadDefaultSettings(
+                testInfo.WorkingPath,
+                configFileName: null,
+                machineWideSettings: new XPlatMachineWideSetting());
+
+            var configSection = settings.GetSection("config");
+            var values = configSection.Items.Select(c => c as AddItem).Where(c => c != null).ToList();
+            var configItems = values.Where(i => i.Key == key);
+            var configFilePath = configItems.Single().ConfigPath;
+
+            // Assert
+            Assert.Equal(0, result.ExitCode);
+            Assert.Equal(value, configItems.Single().Value);
+            Assert.Equal(filePath, configFilePath);
+        }
+
+        [Fact]
+        public void ConfigSetCommand_WithNonExistingConfigSection_AddsConfigSetting()
+        {
+            // Arrange & Act
+            using var testInfo = new TestInfo();
+            var key = "signatureValidationMode";
+            var value = "accept";
+            var filePath = Path.Combine(testInfo.WorkingPath, "NuGet.Config");
+
+            var result = CommandRunner.Run(
+                DotnetCli,
+                Directory.GetCurrentDirectory(),
+                $"{XplatDll} config set {key} {value} --configfile {filePath}",
+                waitForExit: true);
+
+            var settings = Configuration.Settings.LoadDefaultSettings(
```

Comment:
I believe the goal is to not have a `<config>` section, then run the command, then ensure there is a `<config>` section?
How do I know the section didn't exist in the Arrange? From what I see, the command runs, creates the section, then the test loads settings (from DefaultSettings) and ensures the section is there.

Consider this pattern:
1. LoadDefaultSettings 
2.  Read and store the existence of `<config>` section (variable x)
3. Run the command
4.  Read and store the existence of `<config>` section (variable y)
5. Assert x was null/empty
6. Assert y exists